### ISSUE TITLE
Deliver step fails when .ipa file is > 3GB in size. Error: iTMSTransporter could not generate an itmsp: null.

### ISSUE THREAD

**Issue by Kai2k** [fastlane/fastlane#20055]
<!-- Thanks for helping fastlane! Before you submit your issue, please make sure you followed our checklist and check the appropriate boxes by putting an x in the [ ]: [x] -->

### New Issue Checklist

- [x] Updated fastlane to the latest version
- [x] I read the [Contribution Guidelines](https://github.com/fastlane/fastlane/blob/master/CONTRIBUTING.md)
- [x] I read [docs.fastlane.tools](https://docs.fastlane.tools)
- [x] I searched for [existing GitHub issues](https://github.com/fastlane/fastlane/issues)

### Issue Description
Increasing ODR content in our app has meant that our .ipa file is > 3GB in size. This has resulted in the deliver step failing. The error we see is

iTMSTransporter could not generate an itmsp: null

The step works fine when the .ipa file is < 3GB in size.

I notice some similar issues raised in the past (https://github.com/fastlane/fastlane/pull/19697), but our version of fastlane (2.204.3) includes the fixes. I don't see a current issue matching ours.

I raised a previous issue for the same problem which can be found [here](https://github.com/fastlane/fastlane/issues/19849). This issue was closed but the suggested fix did not work for us. We initially reported failures with ipa files > 4GB in size but we've recently seen failures with files 3.2GB in size. 

The failure is consistent - it fails every time. 

We are able to manually upload our ipa using the Transporter Mac app or the ITMS Cli. 

Looking at the docs, we've tried this, in case there were firewall issues - to no avail:

DELIVER_ITMSTRANSPORTER_ADDITIONAL_UPLOAD_PARAMETERS="-t HTTP" fastlane deliver

##### Command executed
Deliver
+--------------------------------------+-----------------------------------------------+
|                               deliver 2.204.3 Summary                                |
+--------------------------------------+-----------------------------------------------+
| api_key                              | ********                                      |
| team_id                              | **********                                    |
| ipa                                  | RELEASE_ARTEFACTS/Dev-AppStore-ClientNameProduct.ipa |
| automatic_release                    | false                                         |
| app_identifier                       | com.ClientNameProduct.ClientNameProduct                     |
| force                                | true                                          |
| skip_metadata                        | true                                          |
| skip_screenshots                     | true                                          |
| skip_app_version_update              | true                                          |
| precheck_include_in_app_purchases    | false                                         |
| screenshots_path                     | ./fastlane/screenshots                        |
| metadata_path                        | ./fastlane/metadata                           |
| app_version                          | x.xx.x                                        |
| platform                             | ios                                           |
| edit_live                            | false                                         |
| use_live_version                     | false                                         |
| skip_binary_upload                   | false                                         |
| overwrite_screenshots                | false                                         |
| sync_screenshots                     | false                                         |
| submit_for_review                    | false                                         |
| reject_if_possible                   | false                                         |
| phased_release                       | false                                         |
| reset_ratings                        | false                                         |
| run_precheck_before_submit           | true                                          |
| precheck_default_rule_level          | warn                                          |
| ignore_language_directory_validation | false                                         |
+--------------------------------------+-----------------------------------------------+

##### Complete output when running fastlane, including the stack trace and command used

<details>
  <pre> + bundle exec fastlane deliver_app_store_build buildType:Dev --env ClientNameProduct
[18:44:51]: Loading from './fastlane/.env.ClientNameProduct'
+--------------------------------------------+---------+----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+
|                                                                                                                                                                                                                 Used plugins                                                                                                                                                                                                                  |
+--------------------------------------------+---------+----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+
| Plugin                                     | Version | Action                                                                                                                                                                                                                                                                                                                                                                                 |
+--------------------------------------------+---------+----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+
| fastlane-plugin-file_manager               | 0.2.0   | remove_files, copy_files                                                                                                                                                                                                                                                                                                                                                               |
| fastlane-plugin-lizard                     | 1.3.3   | lizard                                                                                                                                                                                                                                                                                                                                                                                 |
| fastlane-plugin-brew                       | 0.1.1   | brew                                                                                                                                                                                                                                                                                                                                                                                   |
| fastlane-plugin-ipa_install_plist_generate | 1.0.0   | ipa_install_plist_generate                                                                                                                                                                                                                                                                                                                                                             |
| fastlane-plugin-json                       | 1.1.0   | write_json, download_json, merge_jsons, read_json                                                                                                                                                                                                                                                                                                                                      |
| fastlane-plugin-versioning                 | 0.5.0   | get_version_number_from_git_branch, increment_build_number_in_xcodeproj, get_version_number_from_plist, get_build_number_from_plist, ci_build_number, get_version_number_from_xcodeproj, increment_version_number_in_xcodeproj, increment_version_number_in_plist, get_app_store_version_number, get_info_plist_path, get_build_number_from_xcodeproj, increment_build_number_in_plist |
+--------------------------------------------+---------+----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+

[18:44:53]: -----------------------------
[18:44:53]: --- Step: import_from_git ---
[18:44:53]: -----------------------------
[18:44:53]: Cloning remote git repo...
[18:44:53]: $ git clone git@gitlab.provider.net:back-end-systems/build-pipelines.git /var/folders/sy/47k76vn11dn8tqnpr4369mqc0000gn/T/fl_clone20220311-52639-av55cw/build-pipelines.git --depth 1 --no-checkout 
[18:44:53]: ▸ Cloning into '/var/folders/sy/47k76vn11dn8tqnpr4369mqc0000gn/T/fl_clone20220311-52639-av55cw/build-pipelines.git'...
[18:44:53]: ▸ ################################################################################
[18:44:53]: ▸ #                                host.provider.net                               #
[18:44:53]: ▸ #                  All user logins are monitored and recorded                  #
[18:44:53]: ▸ #        If you are not a provider authorised user, please disconnect now         #
[18:44:53]: ▸ ################################################################################
[18:44:53]: $ cd /var/folders/sy/47k76vn11dn8tqnpr4369mqc0000gn/T/fl_clone20220311-52639-av55cw/build-pipelines.git && git checkout HEAD platform/ios/fastlane/SharedFastfile 
[18:44:53]: ▸ Updated 1 path from 962be67
[18:44:53]: $ cd /var/folders/sy/47k76vn11dn8tqnpr4369mqc0000gn/T/fl_clone20220311-52639-av55cw/build-pipelines.git && git checkout HEAD platform/ios/fastlane/actions
[18:44:53]: ▸ error: pathspec 'platform/ios/fastlane/actions' did not match any file(s) known to git
[18:44:54]: ------------------------------
[18:44:54]: --- Step: default_platform ---
[18:44:54]: ------------------------------
[18:44:54]: Driving the lane 'deliver_app_store_build' 🚀
[18:44:54]: --------------------------------
[18:44:54]: --- Step: ensure_bundle_exec ---
[18:44:54]: --------------------------------
[18:44:54]: Using bundled fastlane ✅
[18:44:54]: ---------------------------------------
[18:44:54]: --- Step: app_store_connect_api_key ---
[18:44:54]: ---------------------------------------
[18:44:54]: ---------------------
[18:44:54]: --- Step: deliver ---
[18:44:54]: ---------------------
[18:44:54]: Creating authorization token for App Store Connect API

+--------------------------------------+-----------------------------------------------+
|                               deliver 2.204.3 Summary                                |
+--------------------------------------+-----------------------------------------------+
| api_key                              | ********                                      |
| team_id                              | **********                                    |
| ipa                                  | RELEASE_ARTEFACTS/Dev-AppStore-ClientNameProduct.ipa |
| automatic_release                    | false                                         |
| app_identifier                       | com.ClientNameProduct.ClientNameProduct                     |
| force                                | true                                          |
| skip_metadata                        | true                                          |
| skip_screenshots                     | true                                          |
| skip_app_version_update              | true                                          |
| precheck_include_in_app_purchases    | false                                         |
| screenshots_path                     | ./fastlane/screenshots                        |
| metadata_path                        | ./fastlane/metadata                           |
| app_version                          | 5.17.3                                        |
| platform                             | ios                                           |
| edit_live                            | false                                         |
| use_live_version                     | false                                         |
| skip_binary_upload                   | false                                         |
| overwrite_screenshots                | false                                         |
| sync_screenshots                     | false                                         |
| submit_for_review                    | false                                         |
| reject_if_possible                   | false                                         |
| phased_release                       | false                                         |
| reset_ratings                        | false                                         |
| run_precheck_before_submit           | true                                          |
| precheck_default_rule_level          | warn                                          |
| ignore_language_directory_validation | false                                         |
+--------------------------------------+-----------------------------------------------+

[18:44:58]: Uploading binary to App Store Connect
[18:45:20]: Going to upload updated app to App Store Connect
[18:45:20]: This might take a few minutes. Please don't interrupt the script.
[18:55:27]: Transporter transfer failed.
[18:55:27]: 
[18:55:27]: iTMSTransporter could not generate an itmsp: null
An exception has occurred: null

[18:55:28]: [iTMSTransporter] 	at com.apple.transporter.core.util.TimeLimitedProcess.call(TimeLimitedProcess.java:146)

[18:55:28]: [iTMSTransporter] 	at com.apple.transporter.core.util.TimeLimitedProcess.call(TimeLimitedProcess.java:25)

[18:55:28]: [iTMSTransporter] 	at java.base/java.util.concurrent.FutureTask.run(Unknown Source)

[18:55:28]: [iTMSTransporter] 	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(Unknown Source)

[18:55:28]: [iTMSTransporter] 	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(Unknown Source)

[18:55:28]: [iTMSTransporter] 	at java.base/java.lang.Thread.run(Unknown Source)

[18:55:28]: [iTMSTransporter] 

[18:55:28]: [iTMSTransporter]  

[18:55:28]: [iTMSTransporter]  

[18:55:28]: [iTMSTransporter]  

[18:55:28]: [iTMSTransporter] Error Summary

[18:55:28]: [iTMSTransporter]  

[18:55:28]: [iTMSTransporter]     iTMSTransporter could not generate an itmsp: null

[18:55:28]: [iTMSTransporter]     An exception has occurred: null

[18:55:28]: [iTMSTransporter] [2022-03-11 18:55:27 GMT] <main> DBG-X: Returning 1

[18:55:28]: iTunes Transporter output above ^
[18:55:28]: iTMSTransporter could not generate an itmsp: null
An exception has occurred: null
Return status of iTunes Transporter was 1: iTMSTransporter could not generate an itmsp: null
\nAn exception has occurred: null
The call to the iTMSTransporter completed with a non-zero exit status: 1. This indicates a failure.
+------------------+-------------------------+
|                Lane Context                |
+------------------+-------------------------+
| ENVIRONMENT      | ClientNameProduct              |
| DEFAULT_PLATFORM | ios                     |
| PLATFORM_NAME    |                         |
| LANE_NAME        | deliver_app_store_build |
+------------------+-------------------------+
[18:55:28]: Error uploading ipa file: 
 

+------+---------------------------+-------------+
|                fastlane summary                |
+------+---------------------------+-------------+
| Step | Action                    | Time (in s) |
+------+---------------------------+-------------+
| 1    | default_platform          | 0           |
| 2    | import_from_git           | 1           |
| 3    | ensure_bundle_exec        | 0           |
| 4    | app_store_connect_api_key | 0           |
| 💥   | deliver                   | 633         |
+------+---------------------------+-------------+

[18:55:28]: fastlane finished with errors

[!] Error uploading ipa file: 
 
script returned exit code 1 </pre>
</details>

### Environment

<!-- 
Please run `fastlane env` and copy the output below. This will help us help you.
If you used the `--capture_output` option, please remove this block as it is already included there. 
-->

<details>
  <pre> <details><summary>✅ fastlane environment ✅</summary>

### Stack

| Key                         | Value                                          |
| --------------------------- | ---------------------------------------------- |
| OS                          | 12.2.1                                         |
| Ruby                        | 2.7.2                                          |
| Bundler?                    | false                                          |
| Git                         | git version 2.30.1 (Apple Git-130)             |
| Installation Source         | ~/.rbenv/versions/2.7.2/bin/fastlane           |
| Host                        | macOS 12.2.1 (21D62)                           |
| Ruby Lib Dir                | ~/.rbenv/versions/2.7.2/lib                    |
| OpenSSL Version             | OpenSSL 1.1.1j  16 Feb 2021                    |
| Is contained                | false                                          |
| Is homebrew                 | false                                          |
| Is installed via Fabric.app | false                                          |
| Xcode Path                  | /Applications/Xcode-13.app/Contents/Developer/ |
| Xcode Version               | 13.0                                           |
| Swift Version               | 5.5                                            |


### System Locale

| Variable | Value       |   |
| -------- | ----------- | - |
| LANG     | en_GB.UTF-8 | ✅ |
| LC_ALL   |             |   |
| LANGUAGE |             |   |


### fastlane files:

<details><summary>`./fastlane/Fastfile`</summary>

```ruby
import_from_git(url: 'git@gitlab.******.net:back-end-systems/build-pipelines.git',
               path: 'platform/ios/fastlane/SharedFastfile')

#import "../../build-pipelines/platform/ios/fastlane/SharedFastfile"

platform :ios do
  	
end

```
</details>

**No Appfile found**


### fastlane gems

| Gem      | Version | Update-Status |
| -------- | ------- | ------------- |
| fastlane | 2.204.3 | ✅ Up-To-Date  |


### Loaded fastlane plugins:

| Plugin                                     | Version | Update-Status |
| ------------------------------------------ | ------- | ------------- |
| fastlane-plugin-file_manager               | 0.2.0   | ✅ Up-To-Date  |
| fastlane-plugin-lizard                     | 1.3.3   | ✅ Up-To-Date  |
| fastlane-plugin-brew                       | 0.1.1   | ✅ Up-To-Date  |
| fastlane-plugin-ipa_install_plist_generate | 1.0.0   | ✅ Up-To-Date  |
| fastlane-plugin-json                       | 1.1.0   | ✅ Up-To-Date  |
| fastlane-plugin-versioning                 | 0.5.0   | ✅ Up-To-Date  |


<details><summary><b>Loaded gems</b></summary>

| Gem                                        | Version      |
| ------------------------------------------ | ------------ |
| did_you_mean                               | 1.4.0        |
| rouge                                      | 2.0.7        |
| xcpretty                                   | 0.3.0        |
| terminal-notifier                          | 2.0.0        |
| terminal-table                             | 1.8.0        |
| plist                                      | 3.6.0        |
| public_suffix                              | 4.0.6        |
| addressable                                | 2.8.0        |
| multipart-post                             | 2.0.0        |
| word_wrap                                  | 1.0.0        |
| optparse                                   | 0.1.1        |
| tty-screen                                 | 0.8.1        |
| tty-cursor                                 | 0.7.1        |
| tty-spinner                                | 0.9.3        |
| artifactory                                | 3.0.15       |
| babosa                                     | 1.0.4        |
| colored                                    | 1.2          |
| highline                                   | 2.0.3        |
| commander                                  | 4.6.0        |
| faraday-cookie_jar                         | 0.0.7        |
| gh_inspector                               | 1.1.3        |
| mini_magick                                | 4.11.0       |
| naturally                                  | 2.2.1        |
| security                                   | 0.1.3        |
| xcpretty-travis-formatter                  | 1.0.1        |
| dotenv                                     | 2.7.6        |
| bundler                                    | 2.1.4        |
| simctl                                     | 1.6.8        |
| uri                                        | 0.10.0       |
| rexml                                      | 3.2.5        |
| nanaimo                                    | 0.3.0        |
| colored2                                   | 3.1.2        |
| claide                                     | 1.0.3        |
| CFPropertyList                             | 3.0.5        |
| atomos                                     | 0.1.3        |
| xcodeproj                                  | 1.21.0       |
| unicode-display_width                      | 1.8.0        |
| excon                                      | 0.91.0       |
| ruby2_keywords                             | 0.0.5        |
| faraday-retry                              | 1.0.3        |
| faraday-rack                               | 1.0.0        |
| faraday-patron                             | 1.0.0        |
| faraday-net_http_persistent                | 1.2.0        |
| faraday-net_http                           | 1.0.1        |
| faraday-multipart                          | 1.0.3        |
| faraday-httpclient                         | 1.0.1        |
| faraday-excon                              | 1.1.0        |
| faraday-em_synchrony                       | 1.0.0        |
| faraday-em_http                            | 1.0.0        |
| faraday                                    | 1.10.0       |
| unf_ext                                    | 0.0.8        |
| unf                                        | 0.1.4        |
| domain_name                                | 0.5.20190701 |
| http-cookie                                | 1.0.4        |
| faraday_middleware                         | 1.2.0        |
| fastimage                                  | 2.2.6        |
| json                                       | 2.6.1        |
| rubyzip                                    | 2.3.2        |
| jwt                                        | 2.3.0        |
| webrick                                    | 1.7.0        |
| httpclient                                 | 2.8.3        |
| multi_json                                 | 1.15.0       |
| signet                                     | 0.16.1       |
| os                                         | 1.1.4        |
| memoist                                    | 0.16.2       |
| googleauth                                 | 1.1.2        |
| mini_mime                                  | 1.1.2        |
| retriable                                  | 3.1.2        |
| trailblazer-option                         | 0.1.2        |
| declarative                                | 0.0.20       |
| uber                                       | 0.1.0        |
| representable                              | 3.1.1        |
| google-apis-core                           | 0.4.2        |
| google-apis-playcustomapp_v1               | 0.7.0        |
| google-apis-androidpublisher_v3            | 0.16.0       |
| rake                                       | 13.0.6       |
| digest-crc                                 | 0.6.4        |
| google-apis-storage_v1                     | 0.11.0       |
| google-apis-iamcredentials_v1              | 0.10.0       |
| google-cloud-errors                        | 1.2.0        |
| google-cloud-env                           | 1.5.0        |
| google-cloud-core                          | 1.6.0        |
| google-cloud-storage                       | 1.36.1       |
| emoji_regex                                | 3.2.3        |
| aws-eventstream                            | 1.2.0        |
| aws-sigv4                                  | 1.4.0        |
| aws-partitions                             | 1.566.0      |
| jmespath                                   | 1.6.1        |
| aws-sdk-core                               | 3.129.1      |
| aws-sdk-kms                                | 1.55.0       |
| aws-sdk-s3                                 | 1.113.0      |
| forwardable                                | 1.3.1        |
| logger                                     | 1.4.2        |
| cgi                                        | 0.1.0        |
| date                                       | 3.0.0        |
| timeout                                    | 0.1.0        |
| stringio                                   | 0.1.0        |
| ipaddr                                     | 1.2.2        |
| openssl                                    | 2.1.2        |
| zlib                                       | 1.1.0        |
| mutex_m                                    | 0.1.0        |
| ostruct                                    | 0.2.0        |
| strscan                                    | 1.0.3        |
| io-console                                 | 0.5.6        |
| delegate                                   | 0.1.0        |
| fileutils                                  | 1.4.1        |
| singleton                                  | 0.1.0        |
| open3                                      | 0.1.0        |
| nokogiri                                   | 1.13.3       |
| racc                                       | 1.6.0        |
| yaml                                       | 0.1.0        |
| psych                                      | 3.1.0        |
| fastlane-plugin-file_manager               | 0.2.0        |
| coderay                                    | 1.1.3        |
| method_source                              | 1.0.0        |
| pry                                        | 0.14.1       |
| fastlane-plugin-lizard                     | 1.3.3        |
| fastlane-plugin-brew                       | 0.1.1        |
| fastlane-plugin-ipa_install_plist_generate | 1.0.0        |
| fastlane-plugin-json                       | 1.1.0        |
| fastlane-plugin-versioning                 | 0.5.0        |
</details>


*generated on:* **2022-03-11**
</details> </pre>
</details>

**Comment by Mehdzor** [comment]
> Hello. I've recently updated all of my gems including fastlane and the outcome is exactly the same. We still receive the same ITMS error. Thanks.

Have you managed to overcome this? Happens for me as well

**Comment by Kai2k** [comment]
@Mehdzor the issue is still on-going for us. 
One avenue that we are investigating is overall time it takes the deliver step to complete. We hypothesise that the Apple api token we are using is going stale before the step has completed. We are investigating if more bandwidth will help. 
I'll report back with any findings.

**Comment by Mehdzor** [comment]
@Kai2k, our app is barely 100Mb, and we experience the same. It happens after binary uploading, and we receive the error almost immediately. It might be something else, but I have no idea.

**Comment by B3ns44d** [comment]
> @Kai2k, our app is barely 100Mb, and we experience the same. It happens after binary uploading, and we receive the error almost immediately. It might be something else, but I have no idea.

we have the same issue with our app

**Comment by Kai2k** [comment]
We are still hoping to test increased bandwidth to see if this cures the issue. I'll report back when done, but there may be a wait as it requires some hardware changes.

**Comment by Kai2k** [comment]
This is still an issue for us. We are in the process of updating some of our hardware, such as the firewall, to increase bandwidth. Once this is done I'll try a fresh build and report back.

**Comment by Kai2k** [comment]
We are still in the process of updating our firewall. I've not forgotten about this and will update if this fixes the issue.

**Comment by Kai2k** [comment]
This is still on-going. Our new firewall should be installed within the next 3-4 weeks. I'll post back after this time.

**Comment by Kai2k** [comment]
Hello. Unfortunately we still have not been able to update our firewall. I've chased this internally and will post the results asap. Thank you.

**Comment by Kai2k** [comment]
Hello again. Good news. This issue is now resolved. The bad news is that I am unsure as to what fixed it. To confirm, we made no specific hardware changes to increase our bandwidth. I can only assume either a Fastlane update, or some back-end update by Apple has resolved this.
